gazebo_plugins/src/gazebo_ros_planar_move.cpp

```diff
@@ -201,20 +285,72 @@ namespace gazebo
     ignition::math::Pose3d pose = parent_->GetWorldPose().Ign();
 #endif
     float yaw = pose.Rot().Yaw();
-    parent_->SetLinearVel(ignition::math::Vector3d(
-          x_ * cosf(yaw) - y_ * sinf(yaw),
-          y_ * cosf(yaw) + x_ * sinf(yaw),
-          0));
-    parent_->SetAngularVel(ignition::math::Vector3d(0, 0, rot_));
+    float dt  = (current_time-this->last_time_).Double();
+
+    ignition::math::Vector3d gravity(parent_->GetWorld()->Gravity());
+    ignition::math::Vector3d linear_vel = ignition::math::Matrix4d(pose.Rot()).Inverse() * parent_->WorldLinearVel();
+    ignition::math::Vector3d angular_vel = parent_->WorldAngularVel();
+
+    // check for joint_state_idel_sec_
+    if ( joint_state_idel_sec_ > 0 &&
+         (current_time-this->last_cmd_subscribe_time_).Double() > joint_state_idel_sec_ ) {
+      if (int((current_time-this->last_cmd_subscribe_time_).Double()*1000)%5000 < 1 ) {
```

Comment:
What do these magic numbers do? What for 5ms?

Reply:
To be honest, I don't know - again, taken from k-okada's fix